pkg/kubernetes/osiris.go

```diff
@@ -34,3 +34,22 @@ func GetMinReplicas(annotations map[string]string, defaultVal int32) int32 {
 	}
 	return int32(minReplicas)
 }
+
+// GetMetricsCheckInterval gets the interval in which the zeroScaler would
+// repeatedly track the pod http request metrics. The value is the number
+// of seconds of the interval. If it fails to do so, it returns the default
+// value instead.
+func GetMetricsCheckInterval(
+	annotations map[string]string,
+	defaultVal int,
+) int {
+	val, ok := annotations["osiris.deislabs.io/metricsCheckInterval"]
+	if !ok {
+		return defaultVal
+	}
+	metricsCheckInterval, err := strconv.Atoi(val)
+	if err != nil {
+		return defaultVal
```

Comment:
This error is silently being swallowed before returning the default value. This will work fine, of course, but could lead to scenarios where an operator wonders _why_ they're getting the default value and why the logs said nothing about it.

My best recommendation here is to wrap and return the error with something like:

```
return Errors.Wrap(err, "error parsing value for annotation osiris.deislabs.io/metricsCheckInterval")
```

Your unit test (thank you, btw, for ensuring there is coverage for this new functionality-- much appreciated) can then add a test case for this error condition-- which is always nice and will improve coverage even further.

The caller can handle the error by logging it and then falling back on the default metrics check interval. The log message should clearly denote why.

Reply:
ok, done. I tried to be consistent with the existing `kubernetes.GetMinReplicas` func, but I totally agree about ignored errors.